openfisca_uk_data/datasets/frs/frs_enhanced/lcf_imputation.py

```diff
@@ -0,0 +1,228 @@
+    spending = MicroDataFrame(spending, weights=spending.weight)
+    emissions.index = emissions.index.astype(str)
+    spending_by_category = spending.groupby("category").spending.sum()
+
+    grouped_ncfs = pd.DataFrame(
+        {
+            code: {
+                "category": CATEGORY_NAMES[code],
+                "carbon_tonnes": emissions[
+                    emissions.index.str.startswith(str(code))
+                ].carbon_tonnes.sum(),
+            }
+            for code in CATEGORY_NAMES.keys()
+        }
+    ).T.set_index("category")
+
+    # For each category, calculate spending from the LCFS and carbon emissions
+    # from the NCFS. Then, divide to find the carbon emissions per pound spent
+
+    carbon_by_category = pd.DataFrame(
+        {
+            category: {
+                "carbon_tonnes": grouped_ncfs.carbon_tonnes[category],
+                "spending": spending_by_category[category],
+            }
+            for category in spending.category.unique()
+        }
+    ).T
+
+    carbon_by_category["carbon_per_pound"] = (
+        carbon_by_category.carbon_tonnes / carbon_by_category.spending
+    )
+
+    # Multiple spending by carbon intensity to calculate LCF households' carbon footprints
+    spending["carbon_tonnes"] = (
+        carbon_by_category.carbon_per_pound[spending.category].values
+        * spending.spending
+    )
+    lcf_with_carbon = (
+        pd.DataFrame(spending[["household", "carbon_tonnes", "weight"]])
+        .groupby("household")
+        .sum()
+    )
+
+    # Add in LCF variables that also appear in the FRS-based microsimulation model
+
+    lcf_with_carbon = pd.concat(
+        [
+            lcf_with_carbon,
+            households[list(HOUSEHOLD_LCF_RENAMES.keys())].rename(
+                columns=HOUSEHOLD_LCF_RENAMES
+            ),
+            people[list(PERSON_LCF_RENAMES) + ["case"]]
+            .rename(columns=PERSON_LCF_RENAMES)
+            .groupby("case")
+            .sum(),
+        ],
+        axis=1,
+    )
+
+    # LCF incomes are weekly - convert to annual
+    for variable in PERSON_LCF_RENAMES.values():
+        lcf_with_carbon[variable] *= 52
+
+    lcf_with_carbon.region = lcf_with_carbon.region.map(REGIONS)
+    lcf = lcf_with_carbon.sort_index()
+
+    # Return household-level LCF dataset with carbon consumption
+    # and FRS-shared columns
+    return MicroDataFrame(lcf, weights=households.weighta[lcf.index] * 1000)
+
+
+def load_and_process_lcf(
+    year: int,
+) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """
+    Load and process the LCF and NCFS summary data.
+
+    Args:
+        year (int): The year of LCFS to use.
+
+    Returns:
+        Tuple[DataFrame, DataFrame, DataFrame]: The LCF household, person and NCFS emissions data.
+    """
+    households = RawLCF.load(2019, "lcfs_2019_dvhh_ukanon")
+    people = RawLCF.load(2019, "lcfs_2019_dvper_ukanon201920")
+    emissions = pd.read_csv(
```

Comment:
Could you move this out of this function, or rename the function? I was confused that `emissions` would be part of `load_and_process_lcf` when it's a separate dataset.

Reply:
Yeah, moved outside thanks